test/unit/controller/test_with_your_users_controller/submit_controller_test.js

```diff
@@ -188,44 +189,117 @@ describe('test with your users - submit controller', () => {
+          supertest(app)
+            .post(paths.prototyping.demoService.confirm)
+            .send(Object.assign({}, VALID_PAYLOAD, {
+              'payment-amount': '£1234.567'
+            }))
+            .end((err, res) => {
+              result = res
+              done(err)
+            })
+        })
+
+        after(() => {
+          nock.cleanAll()
+        })
+
+        it('should redirect with a statusCode of 302', () => {
+          expect(result.statusCode).to.equal(302)
+        })
+        it('should redirect to the create prototype link page', () => {
+          expect(result.header).to.have.property('location').to.equal(paths.prototyping.demoService.create)
+        })
+        it('should add a relevant error message to the session \'flash\'', () => {
+          expect(session.flash).to.have.property('genericError')
+          expect(session.flash.genericError.length).to.equal(1)
+          expect(session.flash.genericError[0]).to.equal('<h2>Use valid characters only</h2> Choose an amount in pounds and pence using digits and a decimal point. For example “10.50”')
+        })
       })
+      describe('because the value exceeds 10,000,000', () => {
+        let result, session, app
+        before('Arrange', () => {
+          nock(CONNECTOR_URL).get(`/v1/frontend/accounts/${GATEWAY_ACCOUNT_ID}`).reply(200, VALID_MINIMAL_GATEWAY_ACCOUNT_RESPONSE)
+          session = getMockSession(VALID_USER)
+          app = createAppWithSession(getApp(), session)
+        })
 
-      it('should add a relevant error message to the session \'flash\'', () => {
-        expect(session.flash).to.have.property('genericError')
-        expect(session.flash.genericError.length).to.equal(1)
-        expect(session.flash.genericError[0]).to.equal('<h2>Use valid characters only</h2> Choose an amount in pounds and pence using digits and a decimal point. For example “10.50”')
+        before('Act', done => {
+          supertest(app)
+            .post(paths.prototyping.demoService.confirm)
+            .send(Object.assign({}, VALID_PAYLOAD, {
+              'payment-amount': '10000000.01'
+            }))
+            .end((err, res) => {
+              result = res
+              done(err)
+            })
+        })
+
+        after(() => {
+          nock.cleanAll()
+        })
+
+        it('should redirect with a statusCode of 302', () => {
+          expect(result.statusCode).to.equal(302)
+        })
+        it('should redirect to the create prototype link page', () => {
+          expect(result.header).to.have.property('location').to.equal(paths.prototyping.demoService.create)
+        })
+        it('should add a relevant error message to the session \'flash\'', () => {
+          expect(session.flash).to.have.property('genericError')
+          expect(session.flash.genericError.length).to.equal(1)
+          expect(session.flash.genericError[0]).to.equal('<h2>Enter a valid amount</h2> Choose an amount under £10,000,000')
```

Comment:
Interesting point here, the assertion of the test relies on the local been setup correctly which I guess we can assume, so no biggy.